main.go

```diff
@@ -344,92 +185,389 @@ func main() {
 	}
-	return errors.New(res.String())
+
+	diffs := rg.ChangedSince(prg)
+
+	of, err := os.Create(*fReportPrefix + "issues.html")
+	if err != nil {
+		return err
+	}
+	log.Print("Creating ", of.Name(), " (this may take a while)...")
+	if err := rg.ReportIssues(of); err != nil {
+		return err
+	}
+	of.Close()
+	filter, err := createFilterFromCmdLine()
+	if err != nil {
+		return err
+	}
+	if !filter.IsEmpty() || diffs != nil {
+		of, err := os.Create(*fReportPrefix + "issues-filtered.html")
+		if err != nil {
+			return err
+		}
+		log.Print("Creating ", of.Name(), " (this may take a while)...")
+		if err := rg.ReportIssuesFiltered(of, &filter, diffs); err != nil {
+			return err
+		}
+		of.Close()
+	}
+
+	return nil
 }
 
-// buildGraph returns the requirements graph at the specified commit, or
-// the graph for the current files if commit is empty. In case the commit
-// is specified, a temporary clone of the repository is created and the
-// path to it is returned.
-func buildGraph(commit string) (*reqGraph, string, error) {
-	if commit == "" {
-		rg, err := CreateReqGraph(*fCertdocPath, *fCodePath)
-		return rg, "", errors.Wrap(err, "Failed to create graph in current dir")
+// reportUp creates a requirements graph (and if necessary for comparison a previous graph) and
+// generates a bottom-up html report, showing the top-level requirement for each implemented function
+// @llr REQ-TRAQ-SWL-35
+func reportUp() error {
+	err := checkCmdLinePaths()
+	if err != nil {
+		return err
 	}
 
-	cwd, err := os.Getwd()
+	var dir string
+	rg, dir, err := buildGraph(*fAt)
 	if err != nil {
-		return nil, "", err
+		return err
 	}
-	dir, err := git.Clone()
+	defer os.RemoveAll(dir)
+
+	var prg *ReqGraph
+	if *fSince != "" {
+		var dir string
+		prg, dir, err = buildGraph(*fSince)
+		if err != nil {
+			return err
+		}
+		defer os.RemoveAll(dir)
+	}
+
+	diffs := rg.ChangedSince(prg)
+
+	of, err := os.Create(*fReportPrefix + "up.html")
 	if err != nil {
-		return nil, dir, err
+		return err
 	}
-	if err = git.Checkout(commit); err != nil {
-		return nil, dir, err
+	log.Print("Creating ", of.Name(), " (this may take a while)...")
+	if err = rg.ReportUp(of); err != nil {
+		return err
 	}
-	rg, err := CreateReqGraph(*fCertdocPath, *fCodePath)
+	of.Close()
+
+	filter, err := createFilterFromCmdLine()
 	if err != nil {
-		return rg, dir, errors.Wrap(err, fmt.Sprintf("Failed to create graph in %s", dir))
+		return err
 	}
-	if err := os.Chdir(cwd); err != nil {
-		return rg, dir, err
+	if !filter.IsEmpty() || diffs != nil {
+		of, err := os.Create(*fReportPrefix + "up-filtered.html")
+		if err != nil {
+			return err
+		}
+		log.Print("Creating ", of.Name(), " (this may take a while)...")
+		if err := rg.ReportUpFiltered(of, &filter, diffs); err != nil {
+			return err
+		}
+		of.Close()
 	}
-	return rg, dir, nil
+
+	return nil
+}
+
+// showHelp prints usage information, either for the root command or for a sub-command
+// @llr REQ-TRAQ-SWL-32
+func showHelp() {
```

Comment:
I think it's consistent now: "reqtraq", "reqtraq help" and "reqtraq --help" will give you the same usage message. I'm not convinced in outputting the defaults though because I think it's a bit confusing not knowing which flags apply to which command.

I'm sure this will be resolved properly when we switch to DBT as you suggest